src/test_distance_functions.cpp

```diff
@@ -0,0 +1,35 @@
+#include <iostream>
+#include "arc_utilities/eigen_helpers.hpp"
+#include "arc_utilities/pretty_print.hpp"
+
+int main(int argc, char** argv)
+{
+    (void)argc;
+    (void)argv;
+
+
+    const Eigen::Matrix3d r1 = Eigen::Matrix3d::Identity();
+    const Eigen::Quaterniond q1(r1);
+
+    for (double theta = -M_PI; theta <= M_PI; theta += M_PI / 20)
+    {
+        Eigen::Quaterniond q2; q2 = Eigen::AngleAxisd(theta, Eigen::Vector3d::UnitX());
+        Eigen::Matrix3d r2 = q2.matrix();
+
+        const double matrix_dist = EigenHelpers::Distance(r1, r2);
+        const double quat_dist = EigenHelpers::Distance(q1, q2);
+        const double eigen_angular_dist = q1.angularDistance(q2);
```

Comment:
since q1.angularDistance(q2) exists, why do we need our own implementation?

Reply:
I don't know specifically why it was written historically. One reason to keep our implementation is for compatibility with existing code. Two potential ways to do that:
 - Keep our implementation as is so that past code runs identically, bit for bit; mark it as deprecated and remove -Werror from CMakeLists.txt
 - Replace the contents with `return q1.angularDistance(q2);`

Reply:
I agree with keeping for compatibility. I was suggesting `return q1.angularDistance(q2)`. Given that you test they are equivalent, might as well just use Eigen's implementation.